### PR TITLE
Add meeting minutes to GitHub Pages

### LINKED ISSUE #17: Add meeting minutes to GitHub pages
Configure Jekyll to allow for easy addition of the weekly meeting minutes.

### PR DESCRIPTION
**Pull request by ianyong**
Changes:
- Site structure
  - Add `minutes` layout to Jekyll.
  - Add index page for meeting minutes that lists all pages using the `minutes` layout automatically.
  - Add link to the meeting minutes index page in the site header.
  - Update GitHub link in the header to point to this repository.
- Dependencies
  - Add `jekyll-spaceship` for greater control over site formatting.
    - Permission requested at nus-cs2103-AY2021S1/forum#267.
  - Remove `github-pages` gem.
    - So as to bypass safe mode and use non-whitelisted plugins.
  - Add previously packaged gems which were part of `github-pages` individually.
    - Disable `jekyll-github-metadata` as we are building the site files manually now.
- Deployment process
  - Add CI job to build the static files for the site before pushing to the orphaned `gh-pages` branch for deployment.
    - See https://github.com/nus-cs2103-AY2021S1/forum/issues/267#issuecomment-698113757 for the rationale behind this.
    - In order to get the generated static files to work with GitHub Pages' URL format, the `baseurl` has to be set as the repo name with a forward slash prepended:
      ```
      bundle exec jekyll build --baseurl /${{ github.event.repository.name }}
      ```
- Documentation
  - Migrate over all meeting minutes thus far.
    - 29 August 2020
    - 5 September 2020
    - 12 September 2020
    - 19 September 2020

To add a new meeting minutes, create a Markdown file in the `/docs/minutes` directory and append the following at the top of the file with the appropriate date substituted in the specified format:
```
---
layout: minutes
date: YYYY-MM-DD
---
```

**Note that for the date, leading zeroes should not be trimmed. Otherwise, sorting on the index page will not work.**

Resolves #17.

### CODE REVIEW

File: .github/workflows/build-jekyll.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Build and Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
```
Comment (wltan) [inline .github/workflows/build-jekyll.yml:6]: Can we verify that this workflow runs smoothly on our repo?

Perhaps you can push an equivalent change directly to a non-master branch first, and see what happens to the `gh-pages` branch.

**Review by wltan — CHANGES_REQUESTED** [review]

File: .github/workflows/build-jekyll.yml (hunk 1 above)
Comment (ianyong) [reply .github/workflows/build-jekyll.yml:6]: I thought that the `GITHUB_TOKEN` secret was being blocked due to this PR being from a fork. Seems to [also happen ](https://github.com/AY2021S1-CS2103T-W16-3/tp/runs/1159335401)with branches on this repo. :/

Comment (ianyong) [reply .github/workflows/build-jekyll.yml:6]: Apparently, `GITHUB_TOKEN` cannot be used; must be a GitHub user's personal access token.

**Comment by ianyong** [comment]
While the current deployment process works with personal access tokens, @wltan brought up the point that we should have it work with the repository token `GITHUB_TOKEN` instead. This is a more ideal solution, especially if Prof Damith were to adopt the use of `jekyll-spaceship`.

**Comment by ianyong** [comment]
Relative links are currently not working. Trying to find a fix.

File: .github/workflows/build-jekyll.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+
+      # Build Jekyll site
+      - name: Build
+        env:
+          JEKYLL_ENV: production
+        run: bundle exec jekyll build --baseurl ${{ github.event.repository.name }} # Set the repo name as the base URL
```
Comment (ianyong) [inline .github/workflows/build-jekyll.yml:42]: Setting the `baseurl` to be the repo name is necessary for GitHub Pages to function.

File: .github/workflows/build-jekyll.yml (hunk 1 above)
Comment (ianyong) [reply .github/workflows/build-jekyll.yml:6]: CI job is able to run now. Tested on a branch of this repo as PRs from forks are unable to access secrets.

See https://github.com/AY2021S1-CS2103T-W16-3/tp/actions/runs/270453191.

Comment (wltan) [reply .github/workflows/build-jekyll.yml:6]: Ok, as seen from https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/65e48f7a1f38c68e643cb29f8d29e0951f06bc4c the CI is able to build and deploy GitHub Pages properly.

**Review by zhaojj2209 — APPROVED** [review]
Tested and LGTM!

**Review by wltan — APPROVED** [review]
Commit https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/8b4e933d0965311b61050847860770722963bb11 resulted in the creation of commit https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/65e48f7a1f38c68e643cb29f8d29e0951f06bc4c on the `gh-pages` branch.

CI run: https://github.com/AY2021S1-CS2103T-W16-3/tp/runs/1160170726

LGTM!

**Pull request merged**